Extensions/Today/TodayViewModel.swift

```diff
@@ -10,20 +10,27 @@ protocol TodayWidgetAppearanceDelegate {
 
 class TodayWidgetViewModel {
     var AppearanceDelegate: TodayWidgetAppearanceDelegate?
-
+    
     func setViewDelegate(todayViewDelegate: TodayWidgetAppearanceDelegate?) {
         self.AppearanceDelegate = todayViewDelegate
     }
 
     func updateCopiedLink() {
-        UIPasteboard.general.asyncURL().uponQueue(.main) { res in
-            guard let url: URL? = res.successValue else {
-                TodayModel.copiedURL = nil
-                self.AppearanceDelegate?.updateCopiedLinkInView(clipboardURL: nil)
+        if !UIPasteboard.general.hasURLs {
+            guard let searchText = UIPasteboard.general.string else {
```

Comment:
This will trigger UI Pasteboard notification on iOS14. 
`UIPasteboard.general.string`

Just making sure thats what you want to do?

If not you could also check if there is *any string in pasteboard* using `UIPasteboard.general.hasStrings` method.

Reply:
@nbhasin2 This is when we read from the pasteboard so it's inevitable you'd get a notification, but it will only start reading when the button is pressed. I'm not just checking if there are strings or not, I'm grabbing them to do a search.